File: src/main/java/cz/vhromada/catalog/gui/game/GamesListDataModel.java
package cz.vhromada.catalog.gui.game;

import java.util.List;

import cz.vhromada.catalog.entity.Game;
import cz.vhromada.catalog.facade.GameFacade;
import cz.vhromada.catalog.gui.commons.AbstractListDataModel;
import cz.vhromada.result.Result;

import org.springframework.util.Assert;

/**
 * A class represents data model for list with games.
 *
 * @author Vladimir Hromada
 */
public class GamesListDataModel extends AbstractListDataModel<Game> {

    /**
     * SerialVersionUID
     */
    private static final long serialVersionUID = 1L;

    /**
     * Facade for games
     */
    private GameFacade gameFacade;

    /**
     * Creates a new instance of GamesListDataModel.
     *
     * @param gameFacade facade for games
     * @throws IllegalArgumentException if facade for games is null
     */
    public GamesListDataModel(final GameFacade gameFacade) {
        Assert.notNull(gameFacade, "Facade for games mustn't be null");

        this.gameFacade = gameFacade;
        update();
    }

    @Override
    protected Result<List<Game>> getData() {
        return gameFacade.getAll();
    }

    @Override
    protected String getDisplayValue(final Game dataObject) {
        return dataObject.getName();
    }

}

File: src/main/kotlin/cz/vhromada/catalog/gui/game/GamesListDataModel.kt
package cz.vhromada.catalog.gui.game

import cz.vhromada.catalog.entity.Game
import cz.vhromada.catalog.facade.GameFacade
import cz.vhromada.catalog.gui.common.AbstractListDataModel
import cz.vhromada.common.result.Result

/**
 * A class represents data model for list with games.
 *
 * @author Vladimir Hromada
 */
class GamesListDataModel(private val gameFacade: GameFacade) : AbstractListDataModel<Game>() {

    init {
        update()
    }

    override fun getData(): Result<List<Game>> {
        return gameFacade.getAll()
    }

    override fun getDisplayValue(dataObject: Game): String {
        return dataObject.name!!
    }

}
